collector/collector/config.py

```diff
@@ -25,9 +25,41 @@ class Config(object):
     SQLALCHEMY_TRACK_MODIFICATIONS = True
     LOG_FILE = 'handler.log'
 
-    # The maximum retention time for records stored in the database, measured
-    # from the time received by the `collector` app.
-    MAX_WEEK_KEEP_RECORDS = 5
+    # When PURGE_OLD_RECORDS == True then a purging system of old records will
+    # be triggered daily. If this variable is not present, then no purging will be done.
+    # If the purging system is enabled, then the following two variables must be set
+    PURGE_OLD_RECORDS = True
+    # The maximum retention time in days for records stored in the database
+    # which do not match the filters in PURGE_FILTERED_RECORDS.
+    # Use 0 to avoid deletion of all unfiltered records.
+    MAX_DAYS_KEEP_UNFILTERED_RECORDS = 30
+    # A dictionary in the following format:
+    # {
+    #     "<field_name>": {
+    #         "<field_value>": <max_days_to_keep>,
+    #         ...
+    #         },
+    #     ...
+    # }
+    # Currently supported fields to filter:
+    # [
+    #     'severity',
+    #     'classification',
+    #     'machine_id'
+    # ]
+    # Use 0 to avoid deletion of records that matches the filter.
+    # If you do not want to filter records to delete, just set an empty dict '{}'
+    PURGE_FILTERED_RECORDS = {
+        "severity": {
+            1: 5,
+            4: 0
+        },
+        "classification": {
+            "org.clearlinux/mce/*": 0,
+            "org.clearlinux/hello/world": 1,
+            "org.clearlinux/heartbeat/ping": 1,
```

Comment:
The `heartbeat/ping` classification is used by the Population view, which includes four charts displaying stats for varying time intervals. I think this purge filter should be removed, unless the Population view is made smarter to not display charts it does not need to. For example, if the filter is set to `7`, the two-week chart and the chart displaying all heartbeat records would be identical to the one-week chart, and thus be confusing to the user.

With that said, if you do think a default filter should exist, then we would need to adjust the Population view accordingly. I think a minimum acceptable default filter would be `7`.

Reply:
ok, lets set these to die in 7 days. (I've heard this before)